Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,328 @@
 }
 
 /**
  @Status Interoperable
 */
-- (const char*)objCType {
-    return "d";
+- (NSNumber*)initWithInteger:(NSInteger)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)decimalNumberWithDecimal:(NSDecimal)decimal {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithLong:(long)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)decimalNumberWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)isNegative {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithShort:(short)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)decimalNumberWithString:(NSString*)numericString locale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedChar:(unsigned char)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)one {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedInt:(unsigned int)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)zero {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedInteger:(NSUInteger)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)notANumber {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedLong:(unsigned long)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithDecimal:(NSDecimal)decimal {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedShort:(unsigned short)value {
+    return [self initWithUnsignedLongLong:static_cast<unsigned long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithUnsignedLongLong:(unsigned long long)value {
+    return [self initWithMantissa:value exponent:0 isNegative:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithString:(NSString*)numericString {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithLongLong:(long long)value {
+    long long val = value;
+    if (val < 0) {
+        val *= -1;
```

Comment:
so given it's LLONG_MIN, the absolute of it will be the same.
then it's being casted into unsigned long long. It matches the reference platform.

---
In reply to: [80504012](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80504012) [](ancestors = 80504012)